### DISCUSSION TITLE
Add stockitem primary key to label

### DISCUSSION THREAD

**Discussion by HannesMorre** [inventree/InvenTree#7243] [Q&A] [answered]
I am once again trying to enable a label printing plugin to our inventree instance. We have failed to do this in past several times.
We updated to 0.14.4 and I am trying to make the label template as we want it, but I cannot get further than getting the qr code and hardcoded text on the label. I am trying to add the stockitem PK on the label. I was assuming this is the "uid" of the Stockitem object, but this does not seem to work.

```
{% extends "label/label_base.html" %}

{% load l10n i18n barcode %}

{% block style %}

.qr {
    position: fixed;
    left: 0mm;
    top: 0mm;
    {% localize off %}
    height: {{ height }}mm;
    width: {{ height }}mm;
    {% endlocalize %}
}

.uid {
    padding: 5px;
    font-weight: bold;
    font-family: Arial, Helvetica, sans-serif;
    height: 100%;
    vertical-align: middle;
    float: right;
    display: inline;
    font-size: 125%;
    position: absolute;
    top: 0mm;
    left: 23mm;
    white-space: nowrap;
    overflow: hidden;
    text-overflow: ellipsis;
}

{% endblock style %}

{% block content %}

<img class="qr" alt="{% trans 'QR Code' %}" src='{% qrcode qr_data %}'>

<div class="uid">
    stockitem: {{ item.uid }}
</div>
{% endblock content %}
```

Can someone please help me out to get at least the stock item id on the label?

**Comment by HannesMorre** [comment]
FYI i tried with just `{{ uid }}` as well instead of `{{ item.uid }}`

**Comment by wolflu05** [comment, marked as answer]
I guess you mean the primary key which is also used in all urls, the barcode, ... you can access it via `item.pk`. Somehow its missing in the docs.

**Reply by HannesMorre** [reply]
okay, didn't think it would be that simple.
Thanks!

**Reply by wolflu05** [reply]
Just FYI, the `uid` field is no longer available, it was migrated to `barcode_hash` since a while. That is populated if you assign a custom barcode (e.g. printed on the stock item from the manufacturer) to the stock item.

**Reply by HannesMorre** [reply]
ah, good to know. Most important for me was the PK for now.
